src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs

```diff
@@ -812,18 +832,16 @@ private bool TryTranslateDateTimeCall(MethodCallExpression node, out BsonValue r
 
             switch (node.Method.Name)
             {
-                case "ToString":
-                    if (node.Arguments.Count == 1)
-                    {
-                        var format = TranslateValue(node.Arguments[0]);
-                        result = new BsonDocument("$dateToString", new BsonDocument
+                case "ToString" when node.Arguments.Count == 1:
+                    var format = TranslateValue(node.Arguments[0]);
+                    result = new BsonDocument("$dateToString", new BsonDocument
                         {
                             { "format", format },
                             { "date", field }
                         });
-                        return true;
-                    }
-                    break;
+                    return true;
+                case "ToString": // without a format argument
+                    return TryTranslateViaToString(node, field, out result);
```

Comment:
I don't think having two cases with the same name ("ToString") is helpful. I think the following more traditional code is easier to understand:

    switch (node.Method.Name)
    {
        case "ToString":
            if (node.Arguments.Count == 1)
            {
                var format = TranslateValue(node.Arguments[0]);
                result = new BsonDocument("$dateToString", new BsonDocument
                {
                    { "format", format },
                    { "date", field }
                });
                return true;
            }
            else
            {
                return TryTranslateViaToString(node, field, out result);
            }
    }

It's also a smaller diff which is often better.

Reply:
done